tensorboard/compat/tensorflow_stub/tensor_manip.py

```diff
@@ -0,0 +1,1005 @@
+        self._message = message
+
+    def __repr__(self):
+        return self._message
+
+
+def _FirstNotNone(l):
+    for x in l:
+        if x is not None:
+            if isinstance(x, ops.Tensor):
+                return _Message("list containing Tensors")
+            else:
+                return x
+    return None
+
+
+def _NotNone(v):
+    if v is None:
+        return _Message("None")
+    else:
+        return v
+
+
+def _FilterTuple(v):
+    if not isinstance(v, (list, tuple)):
+        return v
+    if isinstance(v, tuple):
+        if not any(isinstance(x, (list, tuple)) for x in v):
+            return None
+    if isinstance(v, list):
+        if not any(isinstance(x, (list, tuple)) for x in v):
+            return _FirstNotNone(
+                [None if isinstance(x, (list, tuple)) else x for x in v]
+            )
+    return _FirstNotNone([_FilterTuple(x) for x in v])
+
+
+def _FilterInt(v):
+    if isinstance(v, (list, tuple)):
+        return _FirstNotNone([_FilterInt(x) for x in v])
+    return (
+        None
+        if isinstance(v, (compat.integral_types, tensor_shape.Dimension))
+        else _NotNone(v)
+    )
+
+
+def _FilterFloat(v):
+    if isinstance(v, (list, tuple)):
+        return _FirstNotNone([_FilterFloat(x) for x in v])
+    return None if isinstance(v, compat.real_types) else _NotNone(v)
+
+
+def _FilterComplex(v):
+    if isinstance(v, (list, tuple)):
+        return _FirstNotNone([_FilterComplex(x) for x in v])
+    return None if isinstance(v, compat.complex_types) else _NotNone(v)
+
+
+def _FilterStr(v):
+    if isinstance(v, (list, tuple)):
+        return _FirstNotNone([_FilterStr(x) for x in v])
+    if isinstance(v, compat.bytes_or_text_types):
+        return None
+    else:
+        return _NotNone(v)
+
+
+def _FilterBool(v):
+    if isinstance(v, (list, tuple)):
+        return _FirstNotNone([_FilterBool(x) for x in v])
+    return None if isinstance(v, bool) else _NotNone(v)
+
+
+#
+#
+def _FilterNotTensor(v):
+    if isinstance(v, (list, tuple)):
+        return _FirstNotNone([_FilterNotTensor(x) for x in v])
+    return str(v) if isinstance(v, ops.Tensor) else None
+
+
+#
+#
+_TF_TO_IS_OK = {
+    dtypes.bool: [_FilterBool],
+    dtypes.complex128: [_FilterComplex],
+    dtypes.complex64: [_FilterComplex],
+    dtypes.float16: [_FilterFloat],
+    dtypes.float32: [_FilterFloat],
+    dtypes.float64: [_FilterFloat],
+    dtypes.int16: [_FilterInt],
+    dtypes.int32: [_FilterInt],
+    dtypes.int64: [_FilterInt],
+    dtypes.int8: [_FilterInt],
+    dtypes.qint16: [_FilterInt, _FilterTuple],
+    dtypes.qint32: [_FilterInt, _FilterTuple],
+    dtypes.qint8: [_FilterInt, _FilterTuple],
+    dtypes.quint16: [_FilterInt, _FilterTuple],
+    dtypes.quint8: [_FilterInt, _FilterTuple],
+    dtypes.string: [_FilterStr],
+    dtypes.uint16: [_FilterInt],
+    dtypes.uint8: [_FilterInt],
+}
+#
+#
+def _Assertconvertible(values, dtype):
+    fn_list = _TF_TO_IS_OK.get(dtype, [_FilterNotTensor])
```

Comment:
Let's just update this function so we skip the inoperative _FilterNotTensor entirely - this is the only place it's used so them we can remove the definition above and avoid the ops.Tensor reference inside it, which doesn't work anyway.

```python
def _Assertconvertible(value, dtype):
    # If dtype is None or not recognized, assume it's convertible.
    if dtype is None or dtype not in _TF_TO_IS_OK:
        return
    fn_list = _TF_TO_IS_OK.get(dtype)
    mismatch = _FirstNotNone([fn(values) for fn in fn_list])
    if mismatch is not None:
        raise TypeError(
            "Expected %s, got %s of type '%s' instead."
            % (dtype.name, repr(mismatch), type(mismatch).__name__)
        )
```